tools/jobs.php

```diff
@@ -0,0 +1,427 @@
+     */
+    public function help(): void
+    {
+        $this->usage();
+    }
+
+    /**
+     * Retrieve the columnWidth based on the commands text size
+     */
+    protected function getColumnWidth(array $commands): int
+    {
+        $widths = [];
+
+        foreach ($commands as $command) {
+            $widths[] = Helper::width($command);
+        }
+
+        return $widths ? max($widths) + 2 : 0;
+    }
+
+    /**
+     * List all queued jobs
+     */
+    protected function list(): void
+    {
+        $perPage = $this->getParameterValue('perPage', '10');
+        $page = $this->getParameterValue('page', '1');
+
+        $this->total();
+
+        $data = Repo::job()
+            ->setOutputFormat(Repo::job()::OUTPUT_CLI)
+            ->perPage((int) $perPage)
+            ->setPage((int) $page)
+            ->showQueuedJobs();
+
+        $this->getCommandInterface()
+            ->table(
+                [
+                    [
+                        new TableCell(
+                            __('admin.cli.tool.jobs.queued.jobs.title'),
+                            [
+                                'colspan' => 7,
+                                'style' => new TableCellStyle(['align' => 'center'])
+                            ]
+                        )
+                    ],
+                    [
+                        __('admin.cli.tool.jobs.queued.jobs.fields.id'),
+                        __('admin.cli.tool.jobs.queued.jobs.fields.queue'),
+                        __('admin.cli.tool.jobs.queued.jobs.fields.job.display.name'),
+                        __('admin.cli.tool.jobs.queued.jobs.fields.attempts'),
+                        __('admin.cli.tool.jobs.queued.jobs.fields.reserved.at'),
+                        __('admin.cli.tool.jobs.queued.jobs.fields.available.at'),
+                        __('admin.cli.tool.jobs.queued.jobs.fields.created.at')
+                    ]
+                ],
+                $data->all(),
+            );
+
+        $pagination = [
+            'pagination' => [
+                self::CURRENT_PAGE => $data->currentPage(),
+                self::PREVIOUS_PAGE => ($data->currentPage() - 1) > 0 ? $data->currentPage() - 1 : 1,
+                self::NEXT_PAGE => $data->currentPage(),
+            ],
+        ];
+
+        if ($data->hasMorePages()) {
+            $pagination['pagination'][self::NEXT_PAGE] = $data->currentPage() + 1;
+        }
+
+        $this->getCommandInterface()
+            ->table(
+                [
+                    [
+                        new TableCell(
+                            __('admin.cli.tool.jobs.pagination'),
+                            [
+                                'colspan' => 3,
+                                'style' => new TableCellStyle(['align' => 'center'])
+                            ]
+                        )
+                    ],
+                    [
+                        __('admin.cli.tool.jobs.pagination.current'),
```

Comment:
I think there are implicit keys here (`self::CURRENT_PAGE` et al) -- should those be made explicit?

Reply:
I don't think so, the `self::CURRENT_PAGE` is used on the array constructions, the `i18n` here it's just to display the values according to those.
![Screenshot from 2022-01-20 03-13-48](https://user-images.githubusercontent.com/1243634/150283804-33828f23-a185-4d49-9b4a-b3270c22fe62.png)


And if someone want's to traverse over the results, they need to pass the `--page=` parameter.

Reply:
OK, thanks!